### PR TITLE
Update recipe for py312_patch

### PR DESCRIPTION
**Pull request by cbouss**
Preparation for py312 release.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ requirements:
     - pip
     - nodejs 18.15.0
     - hatch-jupyter-builder 0.8.3
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - hatch-jupyter-builder
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ requirements:
...
-    - hatchling 1.12.2
+    - hatchling 1.18
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - hatchling
```

**Review by skupr-anaconda — APPROVED** [review]
I would remove these pinnings for python packages in host to avoid the same situation for python 3.13 in the future

**Comment by skupr-anaconda** [comment]
Two tests failed on `ppc64le` because of `tornado.simple_httpclient.HTTPTimeoutError: Timeout during request`:
```

FAILED tests/services/kernels/test_cull.py::test_cull_idle[jp_server_config0]
FAILED tests/services/kernels/test_cull.py::test_cull_idle[jp_server_config1]
```
I wonder if after reopening the PR it will succeed

File: recipe/meta.yaml [hunk 3]
```diff
@@ -29,8 +29,8 @@ requirements:
   host:
     - pip
     - nodejs 18.15.0
-    - hatch-jupyter-builder 0.8.3
-    - hatchling 1.12.2
+    - hatch-jupyter-builder
+    - hatchling
     - python
     - setuptools
     - wheel
```
Comment (danpetry) [inline recipe/meta.yaml:36]: `anyio>=3.1.0`, `packaging`

Comment (danpetry) [reply recipe/meta.yaml:36]: `pytest-jupyter-server >=0.4` in test deps

Comment (danpetry) [reply recipe/meta.yaml:36]: https://github.com/jupyter-server/jupyter_server/blob/aeb6f9d0c2c7d1274de8e090d2b4d82cf2e8dde1/pyproject.toml#L29

**Review by danpetry — APPROVED** [review]
A couple of suggestions based on upstream pyproject.toml

File: recipe/meta.yaml (hunk 3 above)
Comment (cbouss) [reply recipe/meta.yaml:36]: The current seems alright to me. pytest-jupyter-server was set to >=0.6 for some reason.

**Pull request merged**